clickhouse-http-client/src/main/java/com/clickhouse/client/http/ClickHouseHttpClient.java

```diff
@@ -28,6 +30,57 @@ public class ClickHouseHttpClient extends AbstractClient<ClickHouseHttpConnectio
 
     static final List<ClickHouseProtocol> SUPPORTED = Collections.singletonList(ClickHouseProtocol.HTTP);
 
+    public static class HostNameAndAddress{
+        public String hostName;
+        public String address;
+    }
+
+    private static HostNameAndAddress getLocalHost() {
+        // get local address but not localhost
+        HostNameAndAddress hostNameAndAddress = new HostNameAndAddress();
+        try {
+            Enumeration<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces();
+
+            for (NetworkInterface ni : Collections.list(networkInterfaces)) {
+                Enumeration<InetAddress> inetAddresses = ni.getInetAddresses();
+                for (InetAddress ia : Collections.list(inetAddresses)) {
+                    // We just use the first non-loopback address
+                    if (!ia.isLoopbackAddress()) {
+                        hostNameAndAddress.address = ia.getHostAddress();
+                        hostNameAndAddress.hostName = ia.getCanonicalHostName();
+                        break;
+                    }
+                }
+            }
+        } catch (SocketException e) {
+            // ignore
+        }
+        return hostNameAndAddress;
+    }
+
+    public static String getReferer(ClickHouseConfig config) {
+        String referer = null;
+        if (!ClickHouseChecker.isNullOrEmpty(config.getStrOption(ClickHouseHttpOption.SEND_HTTP_CLIENT_ID)))
+        {
+            if (ClickHouseChecker.isNullOrEmpty(ClickHouseOption.toKeyValuePairs(config.getStrOption(ClickHouseHttpOption.CUSTOM_HEADERS)).get("referer"))) {
```

Comment:
There is no guarantee that custom header "referer" will be always in lower case, because this map is filed by user and not checked in this way. 
I suggest to move this logic to `createDefaultHeaders` method where it unpacks values and checks if "authorization" header is set (203:196). If done so - the same logic as for "authorization" header can be implemented.